pytorch_lightning/core/lightning.py

```diff
@@ -440,6 +440,20 @@ def __sync(
     def __check_none(name: str, value: Any, _) -> Any:
         raise ValueError(f'`self.log({name}, {value})` was called, but `None` values cannot be logged')
 
+    def log_grad_norm(self, grad_norm_dict: Dict[str, torch.Tensor]) -> None:
```

Comment:
is adding a new hook the only option here? his feels very strange that this only controls the log behavior, and not the calculation. what's the relationship between this and `on_after_bacwkard`?

Reply:
Hey @ananthsub. Users wanted to customise how they aggregate the grad_norm and log them.
Here, they are free to change the reduce_fx to max, or even computed the standard deviation across batches if they want.

`on_after_bacwkard` is independent for this function.